tests/test_interval.py

```diff
@@ -63,6 +63,41 @@ def test_union_via_overloaded_or():
     assert two | one == expected
 
 
+def test_union_via_overloaded_add():
+    one = i.open(3, 6)
+    two = i.open(4, 10)
+    expected = i.open(3, 10)
+    assert one + two == expected
+    assert two + one == expected
+
+
+def test_subtract_non_overlapping():
+    left  = i.open(3, 6)
+    right = i.open(8, 10)
+    expected = i.open(3, 6)
+    assert left - right == expected
+
+
+def test_subtract_overlapping():
+    left  = i.open(3, 6)
+    right = i.open(4, 8)
+    expected = i.openclosed(3, 4)
+    assert left - right == expected
+
+
+def test_subtract_contained():
+    left  = i.open(3, 6)
+    right = i.open(4, 5)
+    expected = IntervalSet([i.openclosed(3, 4), i.closedopen(5, 6)])
+    assert left - right == expected
+
+def test_subtract_almost_complete_overlap():
+    left  = i.closed(1, 2)
+    right = i.open(1, 5)
+    expected = i.closed(1, 1)
+    assert left - right == expected
```

Comment:
what about [1,2] - (1,2)?

or anything that divides the interval should probably break, I think?

(1,4) - (2,3)